tests/test_cargo_alias_config.rs

```diff
@@ -0,0 +1,51 @@
+use support::{project, execs, basic_bin_manifest};
+use hamcrest::{assert_that};
+
+fn setup() {}
+test!(default_alias_config_overrides_config {
+    let p = project("foo")
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"
+            fn main() {
+        }"#)
+        .file(".cargo/config","\
+            [alias]
+            b = \"not_build\"
+        ");;
+
+    assert_that(p.cargo_process("b"),
+                execs().with_status(0).
+                with_stdout_contains("   Compiling [..]"));
+});
+
+test!(alias_config {
+    let p = project("foo")
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"
+            fn main() {
+        }"#)
+        .file(".cargo/config","\
+            [alias]
+            b-cargo-test = \"build\"
+        ");;
+
+    assert_that(p.cargo_process("b-cargo-test"),
+                execs().with_status(0).
+                with_stdout_contains("   Compiling [..]"));
+});
+
+test!(alias_with_flags_config {
+    let p = project("foo")
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file("src/main.rs", r#"
+            fn main() {
+         }"#)
+        .file(".cargo/config","\
+            [alias]
+            b-cargo-test = \"build --release\"
+         ");;
+
+    assert_that(p.cargo_process("b-cargo-test"),
+                execs().with_status(0).
+                with_stdout_contains("   Compiling [..]"));
```

Comment:
It might be worth exhaustively matching here with `-v` to ensure that `-C opt-level` is getting passed